### PR TITLE
Document and Script how to use C++ coverage in CT.

### PR DESCRIPTION
**Pull request by LuisPH3**
This PR is mostly documentation, should explain itself.

But there are some script changes
- enable c++ unit tests coverage in PR CI, because it is pretty fast.
- modify scripts to allow generating reports without compiling or running anything. Useful to compile-run-report arbitrary commands. 

fixes #623

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,6 +32,10 @@ tosca-cpp:
 		-DTOSCA_ASAN="$(TOSCA_CPP_ASAN)"; \
 	cmake --build build --parallel
 
+test-cpp-coverage: TOSCA_CPP_BUILD = Debug
```
Comment (facuMH) [inline Makefile:35]: should this not also be `tosca-cpp-coverage` ?? 

also, should we try to keep it consistent with the other coverage commands, like `ct-coverage-lfvm`. wdyt?

**Review by facuMH — APPROVED** [review]
looks good. just a small naming suggestion

File: Makefile (hunk 1 above)
Comment (LuisPH3) [reply Makefile:35]: I don't think so:

- tosca-cpp: job compiles c++ code
- test-cpp: job compiles and runs c++ unit tests
- test-cpp-coverage: same but with coverage

I will introduce ct-coverage-evmzero in the next pr.

**Comment by LuisPH3** [comment]
fixes  #617

**Review by simonlechner — APPROVED** [review]

**Review by HerbertJordan — COMMENTED** [review]
Waiting for this to be synced with other changes.

**Comment by LuisPH3** [comment]
contents in this file were added to #622

**Pull request closed without merging**